### PR TITLE
Fix some missing acceptable types

### PR DESCRIPTION
**Pull request by kevinmershon**
This pull request handles a number of different issues I have run into while using this library recently:

1. `boolean` values are acceptable inputs for search filters (checkboxes)
1. `FieldValue` and `FieldValue[]` type references are acceptable inputs for use in `search.createFilter` and `search.lookupFields`
1. `Type.DATE` is an acceptable value in `FormatDateTimeOptions`
1. ~~`cache.put` accepts any object, number, or string as value argument~~
1. ~~MapReduce `context.write` accepts any object, number, or string as value argument~~

### CODE REVIEW

File: N/search.d.ts [hunk 1]
```diff
@@ -160,6 +160,8 @@ export interface Search {
     runPaged: SearchRunPagedFunction;
 }
 
+import {FieldValue} from 'Record';
```
Comment (johnogle222) [inline N/search.d.ts]: Could you put the `import` at the top of the file to be consistent with the other uses of this pattern?

**Comment by johnogle222** [comment]
I have very little skin left in this game, but I intentionally forced `cache.put` and `context.write` to take strings as I felt it was a best practice to control your serialization instead of leaving it up to NetSuite to handle. That way you would know for sure what was coming out the other side.

Aside from that I fully support this PR. Thank you!

**Comment by kevinmershon** [comment]
Changes made as you suggest. Ultimately I think you're write about `cache.put` and `context.write`

**Comment by darrenhillconsulting** [comment]
I do agree with John.  I'd rather take care of the serialization myself.
It's predictable.

Force me to a string ... I like it.

**Pull request merged**